### ISSUE TITLE
showQCTests() producing errors for cycles with multiple profiles

### ISSUE THREAD

**Issue by j-harbin** [ArgoCanada/argoFloats#395]
Using the same code as in https://github.com/ArgoCanada/argoFloats/issues/394,

```R
library(argoFloats)
from <- as.POSIXct("2021-02-02", tz="UTC")
to <- as.POSIXct("2021-02-12", tz="UTC")
index1 <- subset(getIndex(), time=list(from=from, to=to))
index2 <- subset(index1, rectangle=list(longitude=c(-68.9232,-58.9485), latitude=c(36.2567,42.5565)))
argos <- readProfiles(getProfiles(index2))
showQCTests(argos[[1]])
```
This works fine, however, `argos[[3]]` is two dimensional (ie. 2 profiles), and the following occurs:

```R
showQCTests(argos[[3]])
```
```R
Error in tests[1, which(action == "QCF$")] : subscript out of bounds
```

As you can see the creation of `qcApp()` has been a great way to detect bugs!

**Comment by dankelley** [comment]
bugs... https://www.youtube.com/watch?v=5utC_KI_SKo

**Comment by j-harbin** [comment]
Dan, maybe you could take a look at this?

I thought it would be the matter of adding [,1] somewhere because using the code above, if I then do:

```R
argos[[3]][['historyAction']]
```
both rows gave the same result:

```R
     [,1]   [,2]  
[1,] "QCP$" "QCF$"
[2,] "QCP$" "QCF$"
```
When I do the following, however,

```R
argos[[3]][["historyQCTest"]]
```
I receive:
```R
     [,1]    [,2]
[1,] "5F03E" "0" 
[2,] "703E"  "0"
```
indicating that maybe different profiles within a cycle pass and/or fail different tests?

**Comment by dankelley** [comment]
I'll take a look.

**Comment by dankelley** [comment]
This is fixed in "develop" commit 60cbbc3732da3fc125a04baadebe66396feca05d.  A reprex follows.  I'll let @j-harbin decide whether to close this.  Note that the formatting is just something that seemed ok to me, with indentation etc.; it's easy to fix that, if it's ugly.

``` r
library(argoFloats)
from <- as.POSIXct("2021-02-02", tz="UTC")
to <- as.POSIXct("2021-02-12", tz="UTC")
index1 <- subset(getIndex(), time=list(from=from, to=to))
#> Kept 4349 profiles (0.18%)
index2 <- subset(index1,
    rectangle=list(longitude=c(-68.9232,-58.9485),
        latitude=c(36.2567,42.5565)))
#> Kept 8 profiles (0.184%)
argos <- readProfiles(getProfiles(index2))
#> Warning in readProfiles(getProfiles(index2)): Of 8 profiles read, 2 have >10% of salinity values with QC flag of 4, signalling bad data.
#>     The indices of the bad profiles are as follows.
#>     2 4
#> Warning in readProfiles(getProfiles(index2)): Of 8 profiles read, 1 has >10% of temperature values with QC flag of 4, signalling bad data.
#>     The indices of the bad profiles are as follows.
#>     4

for (style in c("brief", "full")) {
    for (i in seq_along(argos[["argos"]])) {
        ai <- argos[[i]]
        cat("# ", ai[["filename"]], "\n")
        showQCTests(ai, style)
        cat("\n")
    }
}
#> #  /Users/kelley/Dropbox/data/argo/R4901702_248.nc 
#> Tests performed: 2 3 4 5 6 13 14 15 16 17 19
#>      Passed all tests
#> 
#> #  /Users/kelley/Dropbox/data/argo/R4902102_3210.nc 
#> Tests performed: 2 3 4 5 6 13 14 15 16 17 19
#>      Failed test 15 (Grey List test)
#> 
#> #  /Users/kelley/Dropbox/data/argo/R4902344_164.nc 
#> Profile 1 of 2 profiles
#>     Tests performed: 2 3 4 5 6 13 14 15 16 17 19
#>          Passed all tests
#> Profile 2 of 2 profiles
#>     Tests performed: 2 3 4 5 6 13 14 15
#>          Passed all tests
#> 
#> #  /Users/kelley/Dropbox/data/argo/R4902348_169.nc 
#> Profile 1 of 2 profiles
#>     Tests performed: 2 3 4 5 6 13 14 15 16
#>          Failed test 15 (Grey List test)
#> Profile 2 of 2 profiles
#>     Tests performed: 2 3 4 5 6 13 14 15
#>          Passed all tests
#> 
#> #  /Users/kelley/Dropbox/data/argo/R4902927_117.nc 
#> Profile 1 of 2 profiles
#>     Tests performed: 2 3 4 5 6 13 14 15 16 17 19
#>          Passed all tests
#> Profile 2 of 2 profiles
#>     Tests performed: 2 3 4 5 6 13 14 15
#>          Passed all tests
#> 
#> #  /Users/kelley/Dropbox/data/argo/R4903260_047.nc 
#> Profile 1 of 2 profiles
#>     Tests performed: 2 3 4 5 6 13 14 15 16 17 19
#>          Passed all tests
#> Profile 2 of 2 profiles
#>     Tests performed: 2 3 4 5 6 13 14 15
#>          Passed all tests
#> 
#> #  /Users/kelley/Dropbox/data/argo/R3901604_161.nc 
#> Tests performed: 2 3 4 5 6 7 9 10 13 14 15 16 20 NA
#>      Passed all tests
#> 
#> #  /Users/kelley/Dropbox/data/argo/R4902523_013.nc 
#> Tests performed: 2 3 4 7 8 9 10 11 12 13 14 15 16 17 19 20
#>      Failed test 16 (Gross Salinity or Temperature Sensor Drift test)
#> 
#> #  /Users/kelley/Dropbox/data/argo/R4901702_248.nc 
#>  Test |  Status  | Description
#>  -----|----------|--------------------------------------------------------------
#>     1 |  Skipped | Platform Identification test
#>     2 |   Passed | Impossible Date test
#>     3 |   Passed | Impossible Location test
#>     4 |   Passed | Position on Land test
#>     5 |   Passed | Impossible Speed test
#>     6 |   Passed | Global Range test
#>     7 |  Skipped | Regional Global Parameter test
#>     8 |  Skipped | Pressure Increasing test
#>     9 |  Skipped | Spike test
#>    10 |  Skipped | Top and Bottom Spike test (obsolete)
#>    11 |  Skipped | Gradient test
#>    12 |  Skipped | Digit Rollover test
#>    13 |   Passed | Stuck Value test
#>    14 |   Passed | Density Inversion test
#>    15 |   Passed | Grey List test
#>    16 |   Passed | Gross Salinity or Temperature Sensor Drift test
#>    17 |   Passed | Visual QC test
#>    18 |  Skipped | Frozen profile test
#>    19 |   Passed | Deepest pressure test
#>    20 |  Skipped | Questionable Argos position test
#>    21 |  Skipped | Near-surface unpumped CTD salinity test
#>    22 |  Skipped | Near-surface mixed air/water test
#>    23 |  Skipped | Interim rtqc flag scheme for data deeper than 2000 dbar
#>    24 |  Skipped | Interim rtqc flag scheme for data from experimental sensors
#>    25 |  Skipped | MEDD test
#>  -----|----------|--------------------------------------------------------------
#> 
#> #  /Users/kelley/Dropbox/data/argo/R4902102_3210.nc 
#>  Test |  Status  | Description
#>  -----|----------|--------------------------------------------------------------
#>     1 |  Skipped | Platform Identification test
#>     2 |   Passed | Impossible Date test
#>     3 |   Passed | Impossible Location test
#>     4 |   Passed | Position on Land test
#>     5 |   Passed | Impossible Speed test
#>     6 |   Passed | Global Range test
#>     7 |  Skipped | Regional Global Parameter test
#>     8 |  Skipped | Pressure Increasing test
#>     9 |  Skipped | Spike test
#>    10 |  Skipped | Top and Bottom Spike test (obsolete)
#>    11 |  Skipped | Gradient test
#>    12 |  Skipped | Digit Rollover test
#>    13 |   Passed | Stuck Value test
#>    14 |   Passed | Density Inversion test
#>    15 | *Failed* | Grey List test
#>    16 |   Passed | Gross Salinity or Temperature Sensor Drift test
#>    17 |   Passed | Visual QC test
#>    18 |  Skipped | Frozen profile test
#>    19 |   Passed | Deepest pressure test
#>    20 |  Skipped | Questionable Argos position test
#>    21 |  Skipped | Near-surface unpumped CTD salinity test
#>    22 |  Skipped | Near-surface mixed air/water test
#>    23 |  Skipped | Interim rtqc flag scheme for data deeper than 2000 dbar
#>    24 |  Skipped | Interim rtqc flag scheme for data from experimental sensors
#>    25 |  Skipped | MEDD test
#>  -----|----------|--------------------------------------------------------------
#> 
#> #  /Users/kelley/Dropbox/data/argo/R4902344_164.nc 
#> Profile 1 of 2 profiles
#>      Test |  Status  | Description
#>      -----|----------|--------------------------------------------------------------
#>         1 |  Skipped | Platform Identification test
#>         2 |   Passed | Impossible Date test
#>         3 |   Passed | Impossible Location test
#>         4 |   Passed | Position on Land test
#>         5 |   Passed | Impossible Speed test
#>         6 |   Passed | Global Range test
#>         7 |  Skipped | Regional Global Parameter test
#>         8 |  Skipped | Pressure Increasing test
#>         9 |  Skipped | Spike test
#>        10 |  Skipped | Top and Bottom Spike test (obsolete)
#>        11 |  Skipped | Gradient test
#>        12 |  Skipped | Digit Rollover test
#>        13 |   Passed | Stuck Value test
#>        14 |   Passed | Density Inversion test
#>        15 |   Passed | Grey List test
#>        16 |   Passed | Gross Salinity or Temperature Sensor Drift test
#>        17 |   Passed | Visual QC test
#>        18 |  Skipped | Frozen profile test
#>        19 |   Passed | Deepest pressure test
#>        20 |  Skipped | Questionable Argos position test
#>        21 |  Skipped | Near-surface unpumped CTD salinity test
#>        22 |  Skipped | Near-surface mixed air/water test
#>        23 |  Skipped | Interim rtqc flag scheme for data deeper than 2000 dbar
#>        24 |  Skipped | Interim rtqc flag scheme for data from experimental sensors
#>        25 |  Skipped | MEDD test
#>      -----|----------|--------------------------------------------------------------
#> Profile 2 of 2 profiles
#>      Test |  Status  | Description
#>      -----|----------|--------------------------------------------------------------
#>         1 |  Skipped | Platform Identification test
#>         2 |   Passed | Impossible Date test
#>         3 |   Passed | Impossible Location test
#>         4 |   Passed | Position on Land test
#>         5 |   Passed | Impossible Speed test
#>         6 |   Passed | Global Range test
#>         7 |  Skipped | Regional Global Parameter test
#>         8 |  Skipped | Pressure Increasing test
#>         9 |  Skipped | Spike test
#>        10 |  Skipped | Top and Bottom Spike test (obsolete)
#>        11 |  Skipped | Gradient test
#>        12 |  Skipped | Digit Rollover test
#>        13 |   Passed | Stuck Value test
#>        14 |   Passed | Density Inversion test
#>        15 |   Passed | Grey List test
#>        16 |  Skipped | Gross Salinity or Temperature Sensor Drift test
#>        17 |  Skipped | Visual QC test
#>        18 |  Skipped | Frozen profile test
#>        19 |  Skipped | Deepest pressure test
#>        20 |  Skipped | Questionable Argos position test
#>        21 |  Skipped | Near-surface unpumped CTD salinity test
#>        22 |  Skipped | Near-surface mixed air/water test
#>        23 |  Skipped | Interim rtqc flag scheme for data deeper than 2000 dbar
#>        24 |  Skipped | Interim rtqc flag scheme for data from experimental sensors
#>        25 |  Skipped | MEDD test
#>      -----|----------|--------------------------------------------------------------
#> 
#> #  /Users/kelley/Dropbox/data/argo/R4902348_169.nc 
#> Profile 1 of 2 profiles
#>      Test |  Status  | Description
#>      -----|----------|--------------------------------------------------------------
#>         1 |  Skipped | Platform Identification test
#>         2 |   Passed | Impossible Date test
#>         3 |   Passed | Impossible Location test
#>         4 |   Passed | Position on Land test
#>         5 |   Passed | Impossible Speed test
#>         6 |   Passed | Global Range test
#>         7 |  Skipped | Regional Global Parameter test
#>         8 |  Skipped | Pressure Increasing test
#>         9 |  Skipped | Spike test
#>        10 |  Skipped | Top and Bottom Spike test (obsolete)
#>        11 |  Skipped | Gradient test
#>        12 |  Skipped | Digit Rollover test
#>        13 |   Passed | Stuck Value test
#>        14 |   Passed | Density Inversion test
#>        15 | *Failed* | Grey List test
#>        16 |   Passed | Gross Salinity or Temperature Sensor Drift test
#>        17 |  Skipped | Visual QC test
#>        18 |  Skipped | Frozen profile test
#>        19 |  Skipped | Deepest pressure test
#>        20 |  Skipped | Questionable Argos position test
#>        21 |  Skipped | Near-surface unpumped CTD salinity test
#>        22 |  Skipped | Near-surface mixed air/water test
#>        23 |  Skipped | Interim rtqc flag scheme for data deeper than 2000 dbar
#>        24 |  Skipped | Interim rtqc flag scheme for data from experimental sensors
#>        25 |  Skipped | MEDD test
#>      -----|----------|--------------------------------------------------------------
#> Profile 2 of 2 profiles
#>      Test |  Status  | Description
#>      -----|----------|--------------------------------------------------------------
#>         1 |  Skipped | Platform Identification test
#>         2 |   Passed | Impossible Date test
#>         3 |   Passed | Impossible Location test
#>         4 |   Passed | Position on Land test
#>         5 |   Passed | Impossible Speed test
#>         6 |   Passed | Global Range test
#>         7 |  Skipped | Regional Global Parameter test
#>         8 |  Skipped | Pressure Increasing test
#>         9 |  Skipped | Spike test
#>        10 |  Skipped | Top and Bottom Spike test (obsolete)
#>        11 |  Skipped | Gradient test
#>        12 |  Skipped | Digit Rollover test
#>        13 |   Passed | Stuck Value test
#>        14 |   Passed | Density Inversion test
#>        15 |   Passed | Grey List test
#>        16 |  Skipped | Gross Salinity or Temperature Sensor Drift test
#>        17 |  Skipped | Visual QC test
#>        18 |  Skipped | Frozen profile test
#>        19 |  Skipped | Deepest pressure test
#>        20 |  Skipped | Questionable Argos position test
#>        21 |  Skipped | Near-surface unpumped CTD salinity test
#>        22 |  Skipped | Near-surface mixed air/water test
#>        23 |  Skipped | Interim rtqc flag scheme for data deeper than 2000 dbar
#>        24 |  Skipped | Interim rtqc flag scheme for data from experimental sensors
#>        25 |  Skipped | MEDD test
#>      -----|----------|--------------------------------------------------------------
#> 
#> #  /Users/kelley/Dropbox/data/argo/R4902927_117.nc 
#> Profile 1 of 2 profiles
#>      Test |  Status  | Description
#>      -----|----------|--------------------------------------------------------------
#>         1 |  Skipped | Platform Identification test
#>         2 |   Passed | Impossible Date test
#>         3 |   Passed | Impossible Location test
#>         4 |   Passed | Position on Land test
#>         5 |   Passed | Impossible Speed test
#>         6 |   Passed | Global Range test
#>         7 |  Skipped | Regional Global Parameter test
#>         8 |  Skipped | Pressure Increasing test
#>         9 |  Skipped | Spike test
#>        10 |  Skipped | Top and Bottom Spike test (obsolete)
#>        11 |  Skipped | Gradient test
#>        12 |  Skipped | Digit Rollover test
#>        13 |   Passed | Stuck Value test
#>        14 |   Passed | Density Inversion test
#>        15 |   Passed | Grey List test
#>        16 |   Passed | Gross Salinity or Temperature Sensor Drift test
#>        17 |   Passed | Visual QC test
#>        18 |  Skipped | Frozen profile test
#>        19 |   Passed | Deepest pressure test
#>        20 |  Skipped | Questionable Argos position test
#>        21 |  Skipped | Near-surface unpumped CTD salinity test
#>        22 |  Skipped | Near-surface mixed air/water test
#>        23 |  Skipped | Interim rtqc flag scheme for data deeper than 2000 dbar
#>        24 |  Skipped | Interim rtqc flag scheme for data from experimental sensors
#>        25 |  Skipped | MEDD test
#>      -----|----------|--------------------------------------------------------------
#> Profile 2 of 2 profiles
#>      Test |  Status  | Description
#>      -----|----------|--------------------------------------------------------------
#>         1 |  Skipped | Platform Identification test
#>         2 |   Passed | Impossible Date test
#>         3 |   Passed | Impossible Location test
#>         4 |   Passed | Position on Land test
#>         5 |   Passed | Impossible Speed test
#>         6 |   Passed | Global Range test
#>         7 |  Skipped | Regional Global Parameter test
#>         8 |  Skipped | Pressure Increasing test
#>         9 |  Skipped | Spike test
#>        10 |  Skipped | Top and Bottom Spike test (obsolete)
#>        11 |  Skipped | Gradient test
#>        12 |  Skipped | Digit Rollover test
#>        13 |   Passed | Stuck Value test
#>        14 |   Passed | Density Inversion test
#>        15 |   Passed | Grey List test
#>        16 |  Skipped | Gross Salinity or Temperature Sensor Drift test
#>        17 |  Skipped | Visual QC test
#>        18 |  Skipped | Frozen profile test
#>        19 |  Skipped | Deepest pressure test
#>        20 |  Skipped | Questionable Argos position test
#>        21 |  Skipped | Near-surface unpumped CTD salinity test
#>        22 |  Skipped | Near-surface mixed air/water test
#>        23 |  Skipped | Interim rtqc flag scheme for data deeper than 2000 dbar
#>        24 |  Skipped | Interim rtqc flag scheme for data from experimental sensors
#>        25 |  Skipped | MEDD test
#>      -----|----------|--------------------------------------------------------------
#> 
#> #  /Users/kelley/Dropbox/data/argo/R4903260_047.nc 
#> Profile 1 of 2 profiles
#>      Test |  Status  | Description
#>      -----|----------|--------------------------------------------------------------
#>         1 |  Skipped | Platform Identification test
#>         2 |   Passed | Impossible Date test
#>         3 |   Passed | Impossible Location test
#>         4 |   Passed | Position on Land test
#>         5 |   Passed | Impossible Speed test
#>         6 |   Passed | Global Range test
#>         7 |  Skipped | Regional Global Parameter test
#>         8 |  Skipped | Pressure Increasing test
#>         9 |  Skipped | Spike test
#>        10 |  Skipped | Top and Bottom Spike test (obsolete)
#>        11 |  Skipped | Gradient test
#>        12 |  Skipped | Digit Rollover test
#>        13 |   Passed | Stuck Value test
#>        14 |   Passed | Density Inversion test
#>        15 |   Passed | Grey List test
#>        16 |   Passed | Gross Salinity or Temperature Sensor Drift test
#>        17 |   Passed | Visual QC test
#>        18 |  Skipped | Frozen profile test
#>        19 |   Passed | Deepest pressure test
#>        20 |  Skipped | Questionable Argos position test
#>        21 |  Skipped | Near-surface unpumped CTD salinity test
#>        22 |  Skipped | Near-surface mixed air/water test
#>        23 |  Skipped | Interim rtqc flag scheme for data deeper than 2000 dbar
#>        24 |  Skipped | Interim rtqc flag scheme for data from experimental sensors
#>        25 |  Skipped | MEDD test
#>      -----|----------|--------------------------------------------------------------
#> Profile 2 of 2 profiles
#>      Test |  Status  | Description
#>      -----|----------|--------------------------------------------------------------
#>         1 |  Skipped | Platform Identification test
#>         2 |   Passed | Impossible Date test
#>         3 |   Passed | Impossible Location test
#>         4 |   Passed | Position on Land test
#>         5 |   Passed | Impossible Speed test
#>         6 |   Passed | Global Range test
#>         7 |  Skipped | Regional Global Parameter test
#>         8 |  Skipped | Pressure Increasing test
#>         9 |  Skipped | Spike test
#>        10 |  Skipped | Top and Bottom Spike test (obsolete)
#>        11 |  Skipped | Gradient test
#>        12 |  Skipped | Digit Rollover test
#>        13 |   Passed | Stuck Value test
#>        14 |   Passed | Density Inversion test
#>        15 |   Passed | Grey List test
#>        16 |  Skipped | Gross Salinity or Temperature Sensor Drift test
#>        17 |  Skipped | Visual QC test
#>        18 |  Skipped | Frozen profile test
#>        19 |  Skipped | Deepest pressure test
#>        20 |  Skipped | Questionable Argos position test
#>        21 |  Skipped | Near-surface unpumped CTD salinity test
#>        22 |  Skipped | Near-surface mixed air/water test
#>        23 |  Skipped | Interim rtqc flag scheme for data deeper than 2000 dbar
#>        24 |  Skipped | Interim rtqc flag scheme for data from experimental sensors
#>        25 |  Skipped | MEDD test
#>      -----|----------|--------------------------------------------------------------
#> 
#> #  /Users/kelley/Dropbox/data/argo/R3901604_161.nc 
#>  Test |  Status  | Description
#>  -----|----------|--------------------------------------------------------------
#>     1 |  Skipped | Platform Identification test
#>     2 |   Passed | Impossible Date test
#>     3 |   Passed | Impossible Location test
#>     4 |   Passed | Position on Land test
#>     5 |   Passed | Impossible Speed test
#>     6 |   Passed | Global Range test
#>     7 |   Passed | Regional Global Parameter test
#>     8 |  Skipped | Pressure Increasing test
#>     9 |   Passed | Spike test
#>    10 |   Passed | Top and Bottom Spike test (obsolete)
#>    11 |  Skipped | Gradient test
#>    12 |  Skipped | Digit Rollover test
#>    13 |   Passed | Stuck Value test
#>    14 |   Passed | Density Inversion test
#>    15 |   Passed | Grey List test
#>    16 |   Passed | Gross Salinity or Temperature Sensor Drift test
#>    17 |  Skipped | Visual QC test
#>    18 |  Skipped | Frozen profile test
#>    19 |  Skipped | Deepest pressure test
#>    20 |   Passed | Questionable Argos position test
#>    21 |  Skipped | Near-surface unpumped CTD salinity test
#>    22 |  Skipped | Near-surface mixed air/water test
#>    23 |  Skipped | Interim rtqc flag scheme for data deeper than 2000 dbar
#>    24 |  Skipped | Interim rtqc flag scheme for data from experimental sensors
#>    25 |  Skipped | MEDD test
#>  -----|----------|--------------------------------------------------------------
#> 
#> #  /Users/kelley/Dropbox/data/argo/R4902523_013.nc 
#>  Test |  Status  | Description
#>  -----|----------|--------------------------------------------------------------
#>     1 |  Skipped | Platform Identification test
#>     2 |   Passed | Impossible Date test
#>     3 |   Passed | Impossible Location test
#>     4 |   Passed | Position on Land test
#>     5 |  Skipped | Impossible Speed test
#>     6 |  Skipped | Global Range test
#>     7 |   Passed | Regional Global Parameter test
#>     8 |   Passed | Pressure Increasing test
#>     9 |   Passed | Spike test
#>    10 |   Passed | Top and Bottom Spike test (obsolete)
#>    11 |   Passed | Gradient test
#>    12 |   Passed | Digit Rollover test
#>    13 |   Passed | Stuck Value test
#>    14 |   Passed | Density Inversion test
#>    15 |   Passed | Grey List test
#>    16 | *Failed* | Gross Salinity or Temperature Sensor Drift test
#>    17 |   Passed | Visual QC test
#>    18 |  Skipped | Frozen profile test
#>    19 |   Passed | Deepest pressure test
#>    20 |   Passed | Questionable Argos position test
#>    21 |  Skipped | Near-surface unpumped CTD salinity test
#>    22 |  Skipped | Near-surface mixed air/water test
#>    23 |  Skipped | Interim rtqc flag scheme for data deeper than 2000 dbar
#>    24 |  Skipped | Interim rtqc flag scheme for data from experimental sensors
#>    25 |  Skipped | MEDD test
#>  -----|----------|--------------------------------------------------------------
```

<sup>Created on 2021-02-17 by the [reprex package](https://reprex.tidyverse.org) (v1.0.0)</sup>

**Comment by j-harbin** [comment]
This is great, Dan! Thanks for the quick fix. I checked it on my end as well and all looks good. I'll close this issue.
